core/dbt/parser/manifest.py

```diff
@@ -1280,6 +1288,18 @@ def _check_resource_uniqueness(
 
         alias_resources[full_node_name] = node
 
+    for ver_unver_dict in name_resources.values():
+        versioned_names = ver_unver_dict["ver"].keys()
+        unversioned_names = ver_unver_dict["unver"].keys()
```

Comment:
Non blocking at all, just a suggestion: 

I think it'd be more straightforward to read this method if `name_resources` just have had level of nesting (packages) and omit "ver" and "unver". and then construct the verioned_names and un_versioned names at this point by: 

```python
for name_resources in name_resources_by_package.values():
  versioned_names = {resource.name for resource in name_resources if resource.is_versioned}
  versioned_names = {resource.name for resource in name_resources if not resource.is_versioned}
```

just a little bit less state/nesting to keep in mind while reading.

Reply:
It kind of bothers me that this way of doing it means looping over an n-sized array 3 times instead of 1 though.